tests/pytests/unit/utils/test_cloud.py

```diff
@@ -232,11 +240,32 @@ def test_run_psexec_command_cleanup_lingering_paexec():
         )
         mock_client.return_value.cleanup.assert_called_once()
 
+    # Testing handling an error when it can't delete the PAexec binary
+    with mock_scmr, mock_rm_svc, mock_psexec as mock_client:
+        mock_client.return_value.session = MagicMock(username="Gary")
+        mock_client.return_value.connection = MagicMock(server_name="Krabbs")
+        mock_client.return_value.run_executable.return_value = (
+            "Sandy",
+            "MermaidMan",
+            "BarnicleBoy",
+        )
+        mock_client.return_value.cleanup = MagicMock(side_effect=CannotDelete)
```

Comment:
```suggestion
        mock_client.return_value.cleanup = MagicMock(side_effect=CannotDelete())
```
The side effect should most likely be the class instance, not the class itself.